src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/SimpleChimera.java

```diff
@@ -399,28 +408,30 @@ public SimpleChimera read(final Kryo kryo, final Input input, final Class<Simple
     }
 
     @Override
-    public boolean equals(Object o) {
+    public boolean equals(final Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
 
-        SimpleChimera that = (SimpleChimera) o;
+        final SimpleChimera that = (SimpleChimera) o;
 
         if (isForwardStrandRepresentation != that.isForwardStrandRepresentation) return false;
         if (!sourceContigName.equals(that.sourceContigName)) return false;
         if (!regionWithLowerCoordOnContig.equals(that.regionWithLowerCoordOnContig)) return false;
         if (!regionWithHigherCoordOnContig.equals(that.regionWithHigherCoordOnContig)) return false;
         if (strandSwitch != that.strandSwitch) return false;
-        return insertionMappings.equals(that.insertionMappings);
+        if (!insertionMappings.equals(that.insertionMappings)) return false;
+        return goodNonCanonicalMappingSATag.equals(that.goodNonCanonicalMappingSATag);
     }
 
     @Override
     public int hashCode() {
         int result = sourceContigName.hashCode();
         result = 31 * result + regionWithLowerCoordOnContig.hashCode();
         result = 31 * result + regionWithHigherCoordOnContig.hashCode();
-        result = 31 * result + strandSwitch.ordinal();
+        result = 31 * result + strandSwitch.hashCode();
```

Comment:
Looks like you regenerated this `hashCode` method and reverted from using `ordinal`, which will potentially give rise to bugs as we learned before. Please change back.

Reply:
Oops! Thanks for catching that! auto-generated by Intellij....